core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2400,38 +2407,42 @@ private void freeAndJournal(RpcContext rpcContext, LockedInodePath inodePath, Fr
     long opTimeMs = System.currentTimeMillis();
     List<Inode<?>> freeInodes = new ArrayList<>();
     freeInodes.add(inode);
+    List<LockedInodePath> descendants = mInodeTree.lockDescendants(inodePath,
+        InodeTree.LockMode.WRITE).getInodePathList();
+    Closer closer = Closer.create();
```

Comment:
why does this method still need the closer?

Reply:
Because using a closer is more flexible.  In this particular case, we add the inodePath to the list after  we generate the list, so we don't actually want to close every elements in the array.